packages/block-library/src/embed/edit.js

```diff
@@ -179,6 +180,15 @@ const EmbedEdit = ( props ) => {
 
 	const blockProps = useBlockProps();
 
+	// NOTE: This effect is only required for testing the caption element in the native version,
+	// once the PR is approved this code block should be removed.
+	// Reference: https://github.com/WordPress/gutenberg/pull/32226
+	useEffect( () => {
+		if ( ! url ) {
+			setAttributes( { url: 'http://wordpress.com' } );
+		}
+	}, [] );
+
```

Comment:
If the block doesn't have an URL defined, the HTML related to the caption attribute is not displayed in the HTML mode. Since we don't have yet the ability to set the URL (will be added in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3436), I added this temporary hack to let us test the PR.

Reply:
The Gutenberg side PR of the url-set feature is merged at this point, so wdyt about removing the test code and rely on setting the url as normal?

Reply:
Right, but this code is preventing the embed placeholder to be rendered so we still need it for testing.
https://github.com/WordPress/gutenberg/blob/223c4407ab9c4854648ce10939698a5765d3fa24/packages/block-library/src/embed/edit.js#L225-L230

If I remove this part too then the embed preview component (which contains the caption field) won't be rendered 🙃 , at this point, we could explore a workaround for rendering a fake preview, wdyt?

Reply:
In [this commit](https://github.com/WordPress/gutenberg/pull/32226/commits/93def617478dfa20a9a2a0b93ae629916a4b3d67) I added a workaround for this case, instead of trying the get the embed preview, it's returning the URL as the preview. This way we could have both embed sheet and preview components working together, but I'd like to note that it's just a temporary solution and only for testing, there're other features like replacing the URL that won't work.

Reply:
Ah, I like that "cleaner" intermediate solution with the URL. Let's keep it until the follow up PRs that will wrangle a better preview fallback.